vllm/v1/request.py

```diff
@@ -91,9 +91,11 @@ def append_output_token_ids(
         token_ids: Union[int, List[int]],
     ) -> None:
         if isinstance(token_ids, int):
-            token_ids = [token_ids]
-        self._output_token_ids.extend(token_ids)
-        self._all_token_ids.extend(token_ids)
+            self._output_token_ids.append(token_ids)
+            self._all_token_ids.append(token_ids)
+        else:
+            self._output_token_ids.extend(token_ids)
+            self._all_token_ids.extend(token_ids)
```

Comment:
Is it possible to hide such optimizations, like in `self.output_token_ids.extend(token_ids)` (and `output_token_ids` is a setter)? We have lots of this kinds of optimizations in v0 and they make the code look really complicate, and they are easily reverted by other changes because contributors may mot be aware of this.

Reply:
I see. I've reverted the change.

After second thought and microbenchmarking, I found that this optimization is a bit over-engineering. The overhead from creating the list `[token_ids]` is acceptable. It looks like bigger overhead comes from `isinstance`.